apps/site/assets/ts/schedule/components/LineDiagram.tsx

```diff
@@ -61,9 +79,21 @@ const connectionName = (connection: Route): string => {
 };
 
 const LineDiagram = ({
-  lineDiagram
+  lineDiagram,
+  route,
+  directionId,
+  routePatternsByDirection,
+  services,
+  stops
 }: Props): ReactElement<HTMLElement> | null => {
-  const routeType = lineDiagram[0].route_stop.route!.type;
+  const routeType = route.type;
+  const [modalState, toggleModal] = useState<{
```

Comment:
I think `toggleModal` might be better named `setModalState`, since calling it does not necessarily toggle the modal – you have to provide it with a complete new state, and it wouldn't necessarily change the current open/closed state of the modal.